CorePayments/src/main/java/com/paypal/android/corepayments/api/PatchCCOWithAppSwitchEligibility.kt

```diff
@@ -0,0 +1,108 @@
+package com.paypal.android.corepayments.api
+
+import android.content.Context
+import androidx.annotation.RestrictTo
+import com.paypal.android.corepayments.APIClientError
+import com.paypal.android.corepayments.CoreConfig
+import com.paypal.android.corepayments.RestClient
+import com.paypal.android.corepayments.common.Headers
+import com.paypal.android.corepayments.graphql.GraphQLClient
+import com.paypal.android.corepayments.graphql.GraphQLResult
+import com.paypal.android.corepayments.model.APIResult
+import com.paypal.android.corepayments.model.AppSwitchEligibility
+import com.paypal.android.corepayments.model.ExperimentationContext
+import com.paypal.android.corepayments.model.PatchCcoWithAppSwitchEligibilityRequest
+import com.paypal.android.corepayments.model.TokenType
+import com.paypal.android.corepayments.model.Variables
+import org.json.JSONObject
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class PatchCCOWithAppSwitchEligibility internal constructor(
+    private val authenticationSecureTokenServiceAPI: AuthenticationSecureTokenServiceAPI,
+    private val graphQLClient: GraphQLClient,
+) {
+
+    constructor(coreConfig: CoreConfig) : this(
+        authenticationSecureTokenServiceAPI = AuthenticationSecureTokenServiceAPI(
+            coreConfig,
+            RestClient(coreConfig)
+        ),
+        graphQLClient = GraphQLClient(coreConfig)
+    )
+
+    suspend operator fun invoke(
+        context: Context,
+        orderId: String,
+        tokenType: TokenType,
+        merchantOptInForAppSwitch: Boolean,
+        paypalNativeAppInstalled: Boolean
+    ): APIResult<AppSwitchEligibility> {
+        val tokenResult = authenticationSecureTokenServiceAPI.createLowScopedAccessToken()
+        if (tokenResult is APIResult.Failure) {
+            return APIResult.Failure(tokenResult.error)
```

Comment:
```suggestion
            return tokenResult
```
nit: We could technically forward the result in this case.